test/rules/prefer_intl_args_test.dart

```diff
@@ -0,0 +1,157 @@
+@TestOn('vm')
+
+import 'package:analyzer/dart/analysis/features.dart';
+import 'package:analyzer/dart/analysis/utilities.dart';
+import 'package:dart_code_metrics/src/models/code_issue_severity.dart';
+import 'package:dart_code_metrics/src/rules/prefer_intl_name.dart';
+import 'package:test/test.dart';
+
+// ignore_for_file: use_raw_strings
+
+const _incorrectContent = '''
+import 'package:intl/intl.dart';
+
+final Iterable<String> titles = ['first', 'second'];
+final title = 'first';
+
+class ClassI18n {
+  static final String staticFinalFieldInClassTitle = Intl.message(
+    'static final field in class \$title,
+    name: 'ClassI18n_staticFinalFieldInClassTitle',
+    args: <Object>[],
+  );
+
+  static String get staticPropertyWithExpressionInClassTitle => Intl.message(
+        'static property with expression in class \$title',
+        name: 'ClassI18n_staticPropertyWithExpressionInClassTitle',
+      );
+
+  static String staticMethodExpressionInClassTitle(String title) => Intl.message(
+        'static method with expression in class \$title',
+        name: 'ClassI18n_staticMethodExpressionInClassTitle',
+      );
+
+  final String finalFieldInClassTitle = Intl.message(
+    'final field in class \${titles.map((title) => title)} \$title',
+    name: 'ClassI18n_finalFieldInClassTitle',
+    args: <Object>[titles],
+  );
+
+  String get propertyWithExpressionInClassTitle => Intl.message(
+        'property with expression in class \${titles.map((title) => title)} \$title',
+        name: 'ClassI18n_propertyWithExpressionInClassTitle',
+        args: <Object>[title],
+      );
+}
+''';
+
+const _correctContent = '''
+import 'package:intl/intl.dart';
+
+final Iterable<String> titles = ['first', 'second'];
+final title = 'first';
+
+final String finalFieldTitle = Intl.message(
+  'final field \${titles.map((title) => title)} \$title',
+  name: 'finalFieldTitle',
+  args: <Object>[titles, title],
+);
+
+String get propertyWithExpressionTitle => Intl.message(
+  'property with expression \${titles.map((title) => title)}',
+  name: 'propertyWithExpressionTitle',
+  args: <Object>[titles],
+);
+
+String methodExpressionTitle(String title) => Intl.message(
+  'method with expression \$title',
+  name: 'methodExpressionTitle',
+  args: <Object>[title],
+);
+
+String methodExpressionTitleEmptyArgument() => Intl.message(
+  'method with expression \$title',
+  name: 'methodExpressionTitleEmptyArgument',
+  args: <Object>[title],
+);
+
+String methodExpressionTitleEmptyArgs() => Intl.message(
+  'method with expression',
+  name: 'methodExpressionTitleEmptyArgs',
+  args: <Object>[],
+);
+''';
```

Comment:
many cases is not correct because:
`The purpose of wrapping the message in a function is to allow it to
have parameters which can be used in the result. The message string is
allowed to use a restricted form of Dart string interpolation, where
only the function's parameters can be used, and only in simple
expressions. Local variables cannot be used, and neither can
expressions with curly braces. Only the message string can have
interpolation. The name, desc, args, and examples must be literals and
not contain interpolations. Only the args parameter can refer to
variables, and it should list exactly the function parameters. If you
are passing numbers or dates and you want them formatted, you must do
the formatting outside the function and pass the formatted string into
the message.`

https://github.com/dart-lang/intl/blame/master/README.md#L107

Reply:
Hm, interesting, makes easier to check then

Reply:
@dkrutskikh should we add any warning for that kind of cases?

Reply:
@incendial Why not? I think it's a good idea!

Reply:
I have some work on this rule, I'm ready to make a request today